Remotion/Data/DomainObjects.UnitTests/Mapping/MappingReflectorTest.cs

```diff
@@ -111,13 +113,37 @@ public void Get_WithDuplicateAssembly ()
     }
 
     [Test]
-    public void GetClassDefinitions ()
+    public void GetTypeDefinitions ()
     {
       var assembly = GetType().Assembly;
       var mappingReflector = MappingReflectorObjectMother.CreateMappingReflector(BaseConfiguration.GetTypeDiscoveryService(assembly, assembly));
-      var classDefinitions = mappingReflector.GetTypeDefinitions();
+      var typeDefinitions = mappingReflector.GetTypeDefinitions();
 
-      Assert.That(classDefinitions, Is.Not.Empty);
+      Assert.That(typeDefinitions, Is.Not.Empty);
+    }
+
+    [Test]
+    public void GetTypeDefinitions_DiscoversInterfacesTypes ()
+    {
+      var typeDiscoveryServiceMock = new Mock<ITypeDiscoveryService>();
+      typeDiscoveryServiceMock
+          .Setup(_ => _.GetTypes(typeof(IDomainObject), false))
+          .Returns(Array.Empty<Type>())
+          .Verifiable();
```

Comment:
I'm not 100% sure if that's the best way to test this. We already have a test for GetTypeDefintions, it's just that it only checks for not-empty right now. if we change this to assert that there's 1 or 2 represantative types in there, we could just add 1 or 2 interface types, too, and thus cover the correct usage of the api. Otherwise, we only test that the api was called, but not that it is used correctly.

Reply:
I removed the mock but changed the supplied type discovery service to use a fixed list of types supplied in the tests. This way, the types are specified as nested types in the test fixture. I don't want to add additional types to another test domain that is so far from the actual test. Especially since this test is a unit test.